ssl_checker.py

```diff
@@ -0,0 +1,68 @@
+import socket
+import ssl
+from datetime import datetime, timedelta
+from typing import Dict, Union
+
+import telegram_send
+
+DAYS_TO_EXPIRATION = 50
+SHOW = True
+
+
+def check_cert_validity(cert: Dict) -> bool:
+    not_after = str_to_date(cert['notAfter'])
+    current_time = datetime.utcnow()
+    return current_time + timedelta(days=DAYS_TO_EXPIRATION) <= not_after
+
+
+def str_to_date(date_string: str) -> datetime:
+    return datetime.strptime(date_string.rstrip(' GMT'), "%b %d %H:%M:%S %Y")
+
+
+def get_days_to_expiration(cert: Dict) -> int:
+    not_after = str_to_date(cert['notAfter'])
+    current_date = datetime.utcnow()
+    diff = not_after - current_date
+    return diff.days
+
+
+def send_telegram_notification(days: int, domain: str) -> None:
+    if days >= 0:
+        telegram_send.send(messages=[f'SSL certificate for domain {domain} expires in {days} days'])
+    else:
+        telegram_send.send(messages=[f'SSL certificate for domain {domain} has been expired {abs(days)} days ago'])
+
+
+def get_ssl_cert(target_url: str) -> Union[Dict, None]:
```

Comment:
It's actually great that you know how to use socket library, but it's to low level, for HTTP requests it's better to use requests library, which is the most popular python module and it's good to know it too.